### PR TITLE
write tests for text editing module

### LINKED ISSUE #433: Add tests for TextEditMode module
add tests to NeonCore.test.js for all of the text/bounding box editing features

### PR DESCRIPTION
**Pull request by jacob-hutnyk**
resolves #433

### CODE REVIEW

File: test/DivaBrowser.test.js [hunk 1]
```diff
@@ -41,6 +41,45 @@ describe.each(['firefox', 'chrome', 'safari'])('Tests on %s', (title) => {
     });
   });
 
+  describe('Boundingbox selecting', () => {
+    test('Select syl bbox', async () => {
+      await browser.wait(until.elementLocated(By.id('displayBBox')), 10000);
+      let displayButton = await browser.findElement(By.id('displayBBox'));
+      if (title !== 'safari') {
```
Comment (JRegimbal) [inline test/DivaBrowser.test.js]: Is it absolutely necessary to have different actions for different browsers?

File: test/NeonCore.test.js [hunk 2]
```diff
@@ -38,30 +38,144 @@ beforeAll(() => {
...
+        'elementIds': ['m-2292df83-f3ad-400e-8fc3-4b69b241a30f', 'm-edbf06f6-5791-4d5a-b5c3-2593a2a0eabe']
+      }
+    };
+    expect(await neon.edit(editorAction, pathToPNG)).toBeTruthy();
+    info = await neon.info(pathToPNG);
+    svg = await neon.getSVG(pathToPNG);
+    let arr = info.split(' ');
+    rect1 = svg.getElementById(arr[0]).children[0].children[1];
+    rect2 = svg.getElementById(arr[1]).children[1].children[1];
+    expect(rect1.getAttribute('width')).toBe('278');
```
Comment (JRegimbal) [inline test/NeonCore.test.js]: Where did all of these numbers come from?

File: test/NeonCore.test.js [hunk 3]
```diff
@@ -38,30 +38,144 @@ beforeAll(() => {
...
+    let syl = svg.getElementById('m-369ac5d3-9d2a-4fd3-be4f-cb2a3b530fe5').children[1];
+    let bbox = syl.children[1];
+    expect(bbox.getAttribute('class')).toBe('sylTextRect');
+    let initX = parseInt(bbox.getAttribute('x'));
+    let initY = parseInt(bbox.getAttribute('y'));
+    let editorAction = {
+      'action': 'resize',
+      'param': {
+        'elementId': syl.getAttribute('id'),
+        'ulx': (initX + 100),
```
Comment (JRegimbal) [inline test/NeonCore.test.js]: Put magic numbers into a `const`.

File: test/SinglePageBrowser.test.js [hunk 4]
```diff
@@ -83,6 +83,78 @@ describe.each(['firefox', 'chrome', 'safari'])('Tests on %s', (title) => {
     });
   });
 
+  describe('Check Syl Bounding Box UI', () => {
```
Comment (JRegimbal) [inline test/SinglePageBrowser.test.js]: Is there a reason this wasn't grouped in with other edit mode tests?

File: test/DivaBrowser.test.js (hunk 1 above)
Comment (jacob-hutnyk) [reply test/DivaBrowser.test.js]: safari was not cooperating :( I added a comment before the first if/else

File: test/SinglePageBrowser.test.js (hunk 4 above)
Comment (jacob-hutnyk) [reply test/SinglePageBrowser.test.js]: i kept it separate since it's a separate module in neon and because the beforeAll and afterAll calls are different

**Pull request merged**